src/seedu/addressbook/data/AddressBook.java

```diff
@@ -79,8 +84,17 @@ private void syncTagsWithMasterList(Person person) {
     public void addPerson(Person toAdd) throws DuplicatePersonException {
         syncTagsWithMasterList(toAdd);
         allPersons.add(toAdd);
+        addPersonToTaggingList(toAdd);
     }
 
+    private void addPersonToTaggingList(Person toAdd){
+    	List<Tag> addedTags = toAdd.getTags().getInternalList();
```

Comment:
Your code shows the message we want, but you have missed the concept on Association Class. Hope my previous comments clarifies this.